sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1524,6 +1533,18 @@ def need_ray_up(
                         'Retrying due to list request rate limit exceeded.')
                     return True
 
+                # https://github.com/skypilot-org/skypilot/issues/1797
+                # "The resource 'projects/xxx/zones/us-central1-b/instances/ray-yyy-head-<hash>-compute' was not found" # pylint: disable=line-too-long
+                pattern = (r'\'code\': \'RESOURCE_NOT_FOUND\'.*The resource'
+                           r'.*instances\/.*-compute\' was not found')
+                result = re.search(pattern, stderr)
```

Comment:
nit: should we do a re match or a `json.loads` as the original message seems to be structured message?

Reply:
`stderr` is a bunch of unstructured text lines concat'd from `ray up`, and it includes in the middle
```
...
 raise HttpError(resp, content, uri=self.uri)\ngoogleapiclient.errors.HttpError: <HttpError 404 when requesting https://compute.googleapis.com/compute/v1/projects/skypilot-xxxxx/zones/us-east4-c/instances/ray-dbg-head-44d1e7af-compute?alt=json returned "The resource \'projects/skypilot-xxxxx/zones/us-east4-c/instances/ray-dbg-head-44d1e7af-compute\' was not found". Details: "[{\'message\': "The resource \'projects/skypilot-xxxxx/zones/us-east4-c/instances/ray-dbg-head-44d1e7af-compute\' was not found", \'domain\': \'global\', \'reason\': \'notFound\'}]">\n\
...
```